src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -478,14 +518,17 @@ static int add_partition_xbootldr(DissectedPartition *p) {
                 return 0;
         }
 
-        r = fstab_is_mount_point("/boot");
+        r = slash_boot_in_fstab();
         if (r < 0)
-                return log_error_errno(r, "Failed to parse fstab: %m");
+                return r;
         if (r > 0) {
                 log_debug("/boot specified in fstab, ignoring XBOOTLDR partition.");
                 return 0;
         }
 
+        if (!slash_boot_exists())
+                return 0;
```

Comment:
Did the previous version of the code create the directory? I don't see any provision to create the mount point… I can drop the check obviously, but wouldn't that just mean that the unit would fail if started?

Reply:
Oh, it indeed does create the directory. I didn't know that. I'll rework the patch to ignore a missing directory.

Reply:
the dir is created by the .mount unit, if needed.

this is the only thing that works, because the fs we need to create the dir in might not be writable any earlier then immediately before creating the mount.

Reply:
Yeah. Also the automount unit.